### ISSUE TITLE
Support additional error messages for functions that can fail

### ISSUE THREAD

**Issue by macintacos** [casey/just#1487]
I was using the `env_var()` function, and while it's a nice convenience, I wish I could present the user with some additional information when a variable isn't set (so that they could know what it's for, what to set the value to, valid values that are acceptable, links to documentation, whatever).

I made this request generic, assuming that other functions would likely benefit from this as well.

**Comment by casey** [comment]
I don't think we could do this in a general way, although let me know if anyone has any ideas. We'd have to add an exception handling mechanism, or other generic error handling.

One thing we could do would be to add a second argument to `env_var` functions, which would be a message that was printed if the var isn't found:

```
env_var("foo", "please set $foo")
```

That might be confusing though, since it looks a little like a default, or like it's setting the value.

**Comment by nk9** [comment]
Yeah, having the second parameter be the default is what was proposed in #1278. I'd like to see an example of another scripting language which has a feature like this. TBH, I can't really get my head around how it would be used right now.

**Comment by runeimp** [comment]
New function maybe such as

```
env_or_error("foo", "please set $foo")
```

would be more obvious to how it works?

**Comment by casey** [comment]
Another wacky idea is using doc-comments on assignments, and printing them out if evaluating them fails:

```just
# $FOO should be set to the current port of the foo service
foo :=  env_var("FOO")
```

**Comment by runeimp** [comment]
Maybe the actual error could follow the assignment in a trailing comment and leave normal  doc-comments as simple comments? And simply output the doc-comment on error if a trailing-comment isn't provided? 👼
